azurerm/internal/services/web/app_service_certificate_binding_resource.go

```diff
@@ -0,0 +1,223 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/web/parse"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var appServiceHostnameBindingResourceName = "azurerm_app_service_custom_hostname_binding"
+
+func resourceArmAppServiceCertificateBinding() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServiceCertificateBindingCreate,
+		Read:   resourceArmAppServiceCertificateBindingRead,
+		Delete: resourceArmAppServiceCertificateBindingDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.CertificateBindingID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"hostname_binding_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.HostnameBindingID,
+			},
+
+			"certificate_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CertificateID,
+			},
+
+			"ssl_state": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(web.SslStateIPBasedEnabled),
+					string(web.SslStateSniEnabled),
+				}, false),
+			},
+
+			"hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"app_service_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"thumbprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmAppServiceCertificateBindingCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	certClient := meta.(*clients.Client).Web.CertificatesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for App Service Hostname Binding creation.")
+
+	hostnameBindingID := d.Get("hostname_binding_id").(string)
+	certificateID := d.Get("certificate_id").(string)
+
+	id, err := parse.NewCertificateBindingId(hostnameBindingID, certificateID)
```

Comment:
I'd suggest this wants to pass in a parsed Hostname Binding ID and a parsed Certificate ID, rather than returning the error here?